crates/router/src/core/payments.rs

```diff
@@ -1015,7 +1016,11 @@ where
         .to_validate_request()?
         .validate_request(&req, &merchant_account)?;
 
-    let payment_id = id_type::GlobalPaymentId::generate(state.conf.cell_information.id.clone());
+    // If payment_id is not present in request, generate new payment_id
+    let payment_id = match payment_id {
```

Comment:
This seems misleading. It is hyperswitch that always generates the id. Reading this implies that payment id can also be passed from the request.

Reply:
we can do one thing, generate the payment id from the caller itself